mava/adders/reverb/base.py

```diff
@@ -115,6 +154,214 @@ def __init__(
+                                ):
+                                    new_trajectory.extras[key][
+                                        want_agent
+                                    ] = trajectory.extras[key][cur_agent]
+                                else:
+                                    # TODO: (dries) Only actually need to do this once
+                                    # and not per agent. Maybe fix this in the future.
+                                    new_trajectory.extras[key] = trajectory.extras[key]
+                                if type(trajectory) == mava_types.Transition:
+                                    ext = trajectory.next_extras[key]  # type: ignore
+                                    if (
+                                        type(ext) is dict  # type: ignore
+                                        and cur_agent in ext  # type: ignore
+                                    ):
+                                        new_trajectory.next_extras[key][  # type: ignore
+                                            want_agent
+                                        ] = trajectory.next_extras[  # type: ignore
+                                            key
+                                        ][  # type: ignore
+                                            cur_agent
+                                        ]  # type: ignore
+                                    else:
+                                        # TODO: (dries) Only actually need to
+                                        # do this once and not per agent. Maybe
+                                        # fix this in the future.
+                                        new_trajectory.next_extras[  # type: ignore
+                                            key
+                                        ] = trajectory.next_extras[  # type: ignore
+                                            key
+                                        ]  # type: ignore
+
+                        # Write the new_trajectory to the table.
+                        self._writer.create_item(
+                            table=table, priority=priority, trajectory=new_trajectory
+                        )
+            if not created_item:
+                raise EOFError(
+                    "This experience was not used by any trainer: ",
+                    trajectory.actions.keys(),
+                )
+        else:
+            # Default setting (deprecate this) with only one table. In this setting
+            # we write the entire trajectory to that table.
+            table_items = list(table_priorities.items())
+            assert len(table_items) == 1
+            table, priority = table_items[0]
+            self._writer.create_item(
+                table=table, priority=priority, trajectory=trajectory
+            )
+
+        # Flush the writer.
+        self._writer.flush(self._max_in_flight_items)
```

Comment:
So for certain adders, we previously flushed after every write , e.g. sequence adder  (https://github.com/instadeepai/Mava/blob/cbdbf6a90a3750428f29cc5e893ad67b54d6cee8/mava/adders/reverb/sequence.py#L136). Not we flush the writer once at the end (like we did for the transition adder). Not sure if this has any side-effects?

Reply:
Great observation! I have not considered this. Flush seems to wait for all the data to be written. Please see [here](https://github.com/deepmind/reverb/blob/master/reverb/trajectory_writer.py#L356). So maybe its fine to do this only at the end of a trajectory step. Not sure. I am happy either way. I just don't want to make the writing even slower if we flush to often. But also if we need to flush after each write then we should change it.

Reply:
Sorry, I phrased my previous message incorrectly. The flush is after every `create_item` , not write. 

Let's do a flush after every `create_item`. Acme did that recently across all writers (https://github.com/deepmind/acme/commit/94711b1a59877ebe2d8ce8e79187f7af24ff3472). Please check we do it for the case where it is not multiple trainers as well.

Reply:
Done.

Reply:
I changed this back to how it was previously. I think doing it after every `create_item` broke my training.